N°4325 CMDBSource wrap both mysqli attributes

piRGoif opened the pull request:
In 2.7.5 with N°3513 we added a second mysqli attribute in CMDBSource, so that we can test transactions (see TransactionsTest).

But this wasn't documented, and was really causing confusion.

This refactor wraps both attributes in a dedicated object so that the logic is clearer.

piRGoif committed:
new DbConnectionWrapper

piRGoif committed:
Replace 2 mysqli attr in CMDBSource by the new wrapper

piRGoif committed:
Update autoloader

piRGoif commented:
Tests are showing "Fatal error: Uncaught Error: Class name must be a valid object or a string" errors :/
Will continue work on this...

piRGoif committed:
Fix "Class name must be a valid object or a string" fatal errors
Hahem, no need of a static attribute, we should call the static class instead !

piRGoif commented:
Build is back to normal, PR is ready to review \o/

odain-cbd commented on core/DbConnectionWrapper.php:
```diff
@@ -0,0 +1,52 @@
+
+		return self::$oDbCnxStandard;
+	}
+
```
SetDbConnection is ok like this. but code would be simpler to read if there was two distinct setters. one for mockable object and this one.

piRGoif replied:
Yes, good idea, will do !

piRGoif replied:
Done in e4e7c8fd

odain-cbd reviewed (commented):
PR changes are fine for me. except the composer ones that may be explained by another version of composer?

piRGoif commented:
> 
> 
> PR changes are fine for me. except the composer ones that may be explained by another version of composer?

Thanks for the review !
About the autoloader, this is under discussion in the dev team and will be updated aside

piRGoif committed:
Little refactoring to help readability
* two setters
* getter : add phpdoc
* getter : rename parameter so that the intention is clearer
* rename mock attribute and add phpdoc
Thanks @eespie and @odain for the review !

piRGoif commented:
Forced push to resolve conflicts on the autoloader
Also new commit to add code reviews feedbacks

piRGoif committed:
Rename generic setter
The fact that there are 2 attributes is the responsability of DbConnectionWrapper, the consumer doesn't have to know it
Thanks @eespie for the second review :)

piRGoif committed:
Update parameters that were forgotten

piRGoif committed:
Add the isForQuery parameter everytime we are doing a mysqli::query() call
Not necessary for now but will allow to test other CMDBSource methods without further modifications !

piRGoif committed:
Update DbConnectionWrapper PHPDoc
The since tag were a bit confusing
Thanks @Molkobain for the review !

Molkobain commented:
As discussed with @piRGoif , I'm not confortable with having to manually pass the `$bIsForQuery` parameter when consuming the API (as well as propagating it to other consumers, eg. `CMDBSource::LogDeadlock()`. Ideally we would rework it a bit so that the wrapper automatically returns the right object depending on the called method, but as this should only be used internally we consider it good enough for now.

piRGoif commented:
Merging !
